default/python/universe_generation/empires.py

```diff
@@ -95,26 +85,101 @@ def min_planets_in_vicinity_limit(num_systems):
+                                   for s in systems_pool], reverse=True)
+        current_merit_lower_bound = 0
+
+        best_candidate = []
+
+        # cap the number of attempts to the smaller of the number of systems in the pool, or 100
+        attempts = min(100, len(systems_pool))
+
+        # Cap the number of attempts when the found number of systems is much less than the target
+        # num_home_systems because this indicates that the min_jumps is too large and/or the
+        # systems_pool is too small to ever succeed.
+        num_complete_misses_remaining = 4
+
+        while attempts and num_complete_misses_remaining:
+            # use a local pool of all candidate systems better than the worst threshold merit
+            all_merit_system = [(m, s) for (m, s) in all_merit_system if m > current_merit_lower_bound]
+            local_pool = {s for (m, s) in all_merit_system}
+
+            if len(local_pool) < self.num_home_systems:
+                if len(best_candidate) < self.num_home_systems:
+                    print ("Failing in find_home_systems_for_min_jump_distance because "
+                           "current_merit_lower_bound = {} trims local pool to {} systems "
+                           "which is less than num_home_systems {}.".format(
+                               current_merit_lower_bound, len(local_pool), self.num_home_systems))
+                break
+
+            attempts = min(attempts - 1, len(local_pool))
+
+            candidate = []
+            while local_pool:
+                member = random.choice(list(local_pool))
+                candidate.append(member)
+
+                # remove all neighbors from the local pool
+                local_pool -= set(get_systems_within_jumps(member, min_jumps))
+
+            # Count complete misses when number of candidates is not close to the target.
+            if len(candidate) < (self.num_home_systems - 4):
+                num_complete_misses_remaining -= 1
+
+            if len(candidate) < self.num_home_systems:
+                continue
+
+            # Calculate the merit of the current attempt.  If it is the best so far
+            # keep it and update the merit_threshold
+            merit_system = sorted([(self.system_merit[s], s)
+                                   for s in candidate], reverse=True)[:self.num_home_systems]
+
+            (merit, system) = merit_system[-1]
+
+            # Quit if the lowest merit planet meets the minimum threshold
+            if merit >= min_planets_in_vicinity_limit(get_systems_within_jumps(system, HS_VICINITY_RANGE)):
+                break
+
+            # If we have a better candidate, set the new lower bound one higher to try for a better candidate.
+            if merit >= current_merit_lower_bound:
+                print ("Home system merit lower bound improved from {} to "
+                       "{}".format(current_merit_lower_bound, merit + 1))
+                current_merit_lower_bound = merit + 1
```

Comment:
```merit``` is the lower bound as you know such a combination exists. ```merit+1``` is an arbitrary threshold as you do not know if any combination yields such a value (or higher).

Please do not add the +1 here but change the comparison in the if-operator here to a stricter greater than. If you somehow want to go with the +1, then you absolutely must change the ```all_merit_systems``` fitering (currently in line 136) to a ```>=``` operator. But not adding the +1 here is more precise.